### ISSUE TITLE
Docs: Moving GitHub Wiki content to readthedocs page

### ISSUE THREAD

**Issue by agoscinski** [aiidateam/aiida-core#6400]
### Describe the current issue and solution you'd like

Some Wiki entries for contributors are quite outdated like the [branch description](https://github.com/aiidateam/aiida-core/wiki/Branch-descriptions) therefore I wanted to update them, but was not sure about the current policy so I would prefer to do this within a PR that people can review, so I was thinking to move the content to a `CONTRIBUTING.rst` which can then be seen in the readthedocs. This is also quite standard procedure for a lot of repositories. Does anyone oppose this?
 
Furthermore related but orthogonal to this is the content in the Wiki that seems to be content for the How-To Guides and Topics in the readthedocs. I think it should also be integrated at least checked if it is not already covered. For example there is a page about [plugins](https://github.com/aiidateam/aiida-core/wiki/Testing-plugin-against-latest-AiiDA) in the Wiki that is quite related to the [How to write a plugin for an external code](https://aiida.readthedocs.io/projects/aiida-core/en/latest/howto/plugin_codes.html). Referencing issue https://github.com/aiidateam/aiida-core/issues/4451 that is also about moving old readthedocs content to the current readthedocs. It potentially has slight overlap with the wiki contents, still figuring it out.

**Comment by agoscinski** [comment]
@danielhollas also suggested in https://github.com/aiidateam/aiida-core/pull/6602#issuecomment-2498587516 to just create the internal doc in markdown to simplify the process. This way you can still use the GitHub GUI to edit posts. With sphinx you need to always compile until you can see a preview

**Comment by danielhollas** [comment]
Yeah, I think in general we're kind of overusing Sphinx. In many of the smaller AiiDA(lab) repos (not aiida-core) Sphinx feels like a huge overkill and a chore to setup and maintain. In many cases either just a couple of markdown files, or Mkdocs (which is easier to setup) would suffice.
